common/types/integrations.ts

```diff
@@ -0,0 +1,6 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export type IntegrationConnectionType = 's3' | 'index' | 'securityLake';
```

Comment:
Nit. (not blocking) Some places I see `securityLake` in lower case and some place in upper case. Could be better to keep it consistent.